src/Storages/ObjectStorage/StorageObjectStorage.h

```diff
@@ -175,7 +177,23 @@ class StorageObjectStorage::Configuration
     Configuration() = default;
     virtual ~Configuration() = default;
 
-    using Path = std::string;
+    struct Path
+    {
+        Path() = default;
+        Path(const std::string & path_) : path(path_) {} /// NOLINT(google-explicit-constructor)
+        Path(const std::string & path_, bool allow_partial_prefix_) : path(path_), allow_partial_prefix(allow_partial_prefix_) {}
```

Comment:
```suggestion
        Path(const std::string & path_, bool allow_partial_prefix_ = true) : path(path_), allow_partial_prefix(allow_partial_prefix_) {}
```

Reply:
Also please add a comment about what partial prefix is and hence what allow_partial_prefix means.